com.chartboost.helium/Android~/helium-android-bridge/src/main/kotlin/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,236 @@
+import android.util.Log
+import android.view.Gravity
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.RelativeLayout
+import com.chartboost.heliumsdk.ad.HeliumAd
+import com.chartboost.heliumsdk.ad.HeliumBannerAd
+import com.chartboost.heliumsdk.ad.HeliumBannerAd.HeliumBannerSize
+import com.chartboost.heliumsdk.ad.HeliumFullscreenAd
+import com.chartboost.heliumsdk.ad.HeliumRewardedAd
+import com.unity3d.player.UnityPlayer
+
+class HeliumUnityAdWrapper(private val _ad: HeliumAd?) {
+    // This is the container for the banner.  We need a relative layout to position the banner in one of the 7
+    // possible positions.  HeliumBannerAd is just a FrameLayout.
+    private var mBannerLayout: RelativeLayout? = null
+    private val _activity: Activity = UnityPlayer.currentActivity
+    private val isBanner: Boolean = _ad is HeliumBannerAd
+    private val isFullScreen: Boolean = _ad is HeliumFullscreenAd
+    private var _asBanner: HeliumBannerAd? = null
+    private var _asFullScreen: HeliumFullscreenAd? = null
+    private var startedLoad = false
+
+    private fun ad(): HeliumAd {
+        if (_ad == null) throw RuntimeException("$TAG cannot interact with Helium ad as ad was not created")
+        return _ad
+    }
+
+    init {
+        if (isBanner) _asBanner = _ad as HeliumBannerAd?
+        if (isFullScreen) _asFullScreen = _ad as HeliumFullscreenAd?
+    }
+
+    private fun asBanner(): HeliumBannerAd? {
+        return _asBanner
+    }
+
+    private fun asFullScreen(): HeliumFullscreenAd? {
+        return _asFullScreen
+    }
+
+    fun load() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            ad().load()
+            startedLoad = true
+        }
+    }
+
+    fun load(screenLocation: Int) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            createBannerLayout(screenLocation)
+            load()
+        }
+    }
+
+    fun show() {
+        HeliumUnityBridge.runTaskOnUiThread { if (isFullScreen) asFullScreen()!!.show() }
+    }
+
+    fun setKeyword(keyword: String?, value: String?): Boolean {
+        val keywords = ad().keywords // not null by design
+        return keywords.set(keyword!!, value!!)
+    }
+
+    fun removeKeyword(keyword: String?): String? {
+        val keywords = ad().keywords // not null by design
+        return keywords.remove(keyword!!)
+    }
+
+    fun setCustomData(customData: String?) {
+        val ad = ad()
+        if (ad is HeliumRewardedAd) {
+            ad.customData = customData
+        } else {
+            throw RuntimeException("$TAG custom data can only be set on a rewarded ad")
+        }
+    }
+
+    fun setBannerVisibility(isVisible: Boolean) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            if (isBanner && mBannerLayout != null) {
+                val visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
+                mBannerLayout!!.visibility = visibility
+                asBanner()!!.visibility = visibility
```

Comment:
Don't use `!!` if we can.

Reply:
Correct, use ?, this is from automatic conversion and will keep it in mind from now on :)